src/domain_bridge/compress_messages.cpp

```diff
@@ -0,0 +1,93 @@
+#include "domain_bridge/compress_messages.hpp"
+
+#include <zstd.h>
+
+#include <stdexcept>
+#include <sstream>
+#include <vector>
+
+namespace domain_bridge
+{
+
+// Increasing the compression level will:
+//   - Increase the time taken to compress
+//   - Decrease the size of the compressed data
+// Setting to zero uses Zstd's default value of 3.
+constexpr const int kDefaultZstdCompressionLevel = 1;
```

Comment:
It might make sense to expose this as a option. Can be done in a follow-up PR.

Reply:
Yes, I prefer doing it that in a follow-up.
The currently harcoded compression level works reasonably well.